library/spdm_requester_lib/libspdm_req_get_measurement_extension_log.c

```diff
@@ -0,0 +1,307 @@
+            spdm_context, true, 0,
+            SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEL_CAP)) {
+        return LIBSPDM_STATUS_UNSUPPORTED_CAP;
+    }
+    if (spdm_context->connection_info.connection_state < LIBSPDM_CONNECTION_STATE_NEGOTIATED) {
+        return LIBSPDM_STATUS_INVALID_STATE_LOCAL;
+    }
+
+    session_info = NULL;
+    if (session_id != NULL) {
+        session_info = libspdm_get_session_info_via_session_id(spdm_context, *session_id);
+        if (session_info == NULL) {
+            LIBSPDM_ASSERT(false);
+            return LIBSPDM_STATUS_INVALID_STATE_LOCAL;
+        }
+        session_state = libspdm_secured_message_get_session_state(
+            session_info->secured_message_context);
+        if (session_state != LIBSPDM_SESSION_STATE_ESTABLISHED) {
+            return LIBSPDM_STATUS_INVALID_STATE_LOCAL;
+        }
+    }
+
+    libspdm_reset_message_buffer_via_request_code(spdm_context, session_info,
+                                                  SPDM_GET_MEASUREMENT_EXTENSION_LOG);
+
+    chunk_enabled =
+        libspdm_is_capabilities_flag_supported(spdm_context, true,
+                                               SPDM_GET_CAPABILITIES_REQUEST_FLAGS_CHUNK_CAP,
+                                               SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHUNK_CAP);
+
+    remainder_length = 0;
+    total_responder_mel_buffer_length = 0;
+    mel_capacity = *mel_size;
+    mel_size_internal = 0;
+
+    transport_header_size = spdm_context->local_context.capability.transport_header_size;
+
+    do {
+        /* -=[Construct Request Phase]=- */
+        status = libspdm_acquire_sender_buffer (spdm_context, &message_size, (void **)&message);
+        if (LIBSPDM_STATUS_IS_ERROR(status)) {
+            return status;
+        }
+        LIBSPDM_ASSERT (message_size >= transport_header_size +
+                        spdm_context->local_context.capability.transport_tail_size);
+        spdm_request = (void *)(message + transport_header_size);
+        spdm_request_size = message_size - transport_header_size -
+                            spdm_context->local_context.capability.transport_tail_size;
+
+        spdm_request->header.spdm_version = libspdm_get_connection_version (spdm_context);
+        spdm_request->header.request_response_code = SPDM_GET_MEASUREMENT_EXTENSION_LOG;
+        spdm_request->header.param1 = 0;
+        spdm_request->header.param2 = 0;
+        spdm_request->offset = (uint32_t)mel_size_internal;
+        if (spdm_request->offset == 0) {
+            spdm_request->length = length;
+        } else {
+            spdm_request->length = LIBSPDM_MIN(length, remainder_length);
+        }
+        spdm_request_size = sizeof(spdm_get_measurement_extension_log_request_t);
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "request (offset 0x%x, size 0x%x):\n",
+                       spdm_request->offset, spdm_request->length));
+
+        LIBSPDM_ASSERT(spdm_request->header.spdm_version >= SPDM_MESSAGE_VERSION_13);
```

Comment:
This should be moved to the Verify State Phase as a runtime check. Ie
```
    if (libspdm_get_connection_version(spdm_context) < SPDM_MESSAGE_VERSION_13) {
        return LIBSPDM_STATUS_UNSUPPORTED_CAP;
    }
```

Reply:
Thanks. I have moved the check to verify state phase.